ntp-proto/src/time_types.rs

```diff
@@ -381,6 +382,50 @@ ntp_duration_scalar_div!(u16);
 ntp_duration_scalar_div!(u32);
 // u64 and usize deliberately excluded as they can result in overflows
 
+/// Stores when we will next exchange packages with a remote server.
+///
+/// The value is in seconds stored in log2 format:
+///
+/// - a value of 4 means 2^4 = 16 seconds
+/// - a value of 17 is 2^17 = ~36h
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PollInterval(i8);
+
+impl PollInterval {
+    pub const MIN: Self = Self(4);
+    pub const MAX: Self = Self(17);
+
+    #[must_use]
+    pub fn inc(self) -> Self {
+        Self(self.0 + 1).min(Self::MAX)
+    }
+
+    #[must_use]
+    pub fn dec(self) -> Self {
+        Self(self.0 - 1).max(Self::MIN)
+    }
```

Comment:
Not sure I agree with the change in interface here. This is going to result in a lot of code following the pattern `a=a.inc()`, which is why I originally decided on modifying inc/dec